hpcbench/toolbox/contextlib_ext.py

```diff
@@ -102,3 +102,35 @@ def mkdtemp(*args, **kwargs):
     finally:
         if remove:
             shutil.rmtree(path)
+
+
+@contextlib.contextmanager
+def modified_environ(*remove, **update):
+    """
+    Temporarily updates the ``os.environ`` dictionary in-place.
+
+    The ``os.environ`` dictionary is updated in-place so that the modification
+    is sure to work in all situations.
+
+    :param remove: Environment variables to remove.
+    :param update: Dictionary of environment variables
+    and values to add/update.
+    """
+    env = os.environ
+    update = update or {}
+    remove = remove or []
+
+    # List of environment variables being updated or removed.
+    stomped = (set(update.keys()) | set(remove)) & set(env.keys())
```

Comment:
<nitpick>`set(update)` would also do </nitpick>

Reply:
you're right